rust/src/ikev1/ikev1.rs

```diff
@@ -0,0 +1,606 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Frank Honza <frank.honza@dcso.de>
+
+use std;
+use crate::core::{self, ALPROTO_UNKNOWN, AppProto, Flow, STREAM_TOSERVER, STREAM_TOCLIENT};
+use crate::log::*;
+use std::mem::transmute;
+use crate::applayer;
+use crate::applayer::*;
+use std::ffi::CString;
+use nom;
+use crate::ikev1::parser::*;
+use std::collections::HashSet;
+
+static mut ALPROTO_IKEV1: AppProto = ALPROTO_UNKNOWN;
+
+pub struct IKEV1Transaction {
+    tx_id: u64,
+    progress: i32,
+    pub request: Option<String>,
+    pub response: Option<String>,
+
+    pub spi_initiator: Option<u64>,
+    pub spi_responder: Option<u64>,
+    pub maj_ver: Option<u8>,
+    pub min_ver: Option<u8>,
+    pub exchange_type: Option<u8>,
+    pub payload_types: Option<HashSet<u8>>,
+    pub encrypted_payloads: bool,
+
+    logged: LoggerFlags,
+    de_state: Option<*mut core::DetectEngineState>,
+    events: *mut core::AppLayerDecoderEvents,
+    detect_flags: applayer::TxDetectFlags,
+}
+
+impl IKEV1Transaction {
+    pub fn new() -> IKEV1Transaction {
+        IKEV1Transaction {
```

Comment:
nice, didn't know the Default impl, for the state this works fine but the transaction requires applayer::LoggerFlags and core::AppLayerDecoderEvents. For the LoggerFlags this could be derived/implemented but with the AppLayerDecoderEvents i'm not sure. (defined as an empty enum)